### PR TITLE
include, os: s/ctz/std::countr_zero/

### PR DESCRIPTION
**Pull request by tchaikov**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by Matan-B** [comment]
Can we omit `include "include/intarith.h"` in `denc.h`?

**Comment by tchaikov** [comment]
> Can we omit `include "include/intarith.h"` from `denc.h`?

@Matan-B yes! thanks for pointing this out. removed and repushed.

**Review by Matan-B — APPROVED** [review]

**Comment by tchaikov** [comment]
changelog

- include `include/intarith.h` in `histogram.h` as the latter uses `cbits()`, as we dropped `#include "include/intarith.h"` in `denc.h` which is in turn included by `histogram.h`, so we have to include `include/intarith.h` in `histogram.h` after this change.

**Comment by tchaikov** [comment]
jenkins test api

**Review by cbodley — APPROVED** [review]

**Comment by tchaikov** [comment]
jenkins test api

**Comment by tchaikov** [comment]
https://pulpito.ceph.com/kchai-2022-08-30_00:38:22-rados-wip-kefu-testing-2022-08-29-1815-distro-default-smithi/

failures tracked by https://tracker.ceph.com/issues/57332

**Comment by tchaikov** [comment]
jenkins test api

File: src/include/denc.h [hunk 1]
```diff
@@ -514,7 +514,7 @@ inline void denc_signed_varint_lowz(int64_t v, It& p) {
     v = -v;
     negative = true;
   }
-  unsigned lowznib = v ? (ctz(v) / 4) : 0u;
+  unsigned lowznib = v ? (std::countr_zero(std::bit_cast<uint64_t>(v)) / 4) : 0u;
```
Comment (yaarith) [inline src/include/denc.h:517]: hi @tchaikov, can you please take a look at https://tracker.ceph.com/issues/57355?
Looks like this change is failing the windows tests.

Comment (tchaikov) [reply src/include/denc.h:517]: sure, i am staring at it.

**Pull request merged**